osu.Game/Rulesets/Difficulty/Skills/Skill.cs

```diff
@@ -58,6 +61,21 @@ public void Process(DifficultyHitObject current)
             Previous.Push(current);
         }
 
+        public void TouchProcess(DifficultyHitObject current)
+        {
+            var dummy = prevDummyStrain;
+            dummyStrain *= strainDecay(current.DeltaTime);
+            dummyStrain += StrainValueOf(current) * SkillMultiplier;
+
+            prevDummyStrain = previous * strainDecay(current.DeltaTime);
+            prevDummyStrain += StrainValueOf(current) * SkillMultiplier;
+            currentStrain = (prevDummyStrain + dummyStrain) / 2;
+            previous = dummy;
```

Comment:
I'd like comments to be added to this method explaining what this process does. I.e. why it's using the last-last object strain, and why the averaging is done above.

Reply:
The old values used to only use the last-last object strain instead of an average due to how touchscreen generally used two cursors. However, after further discussion with touchscreen players, we have found that it would make the most sense to average it to dilute the overall results. After further discussion, we have found that it would make the most sense to revert this completely. This would in turn, resolve many of the issues that derive from this.

Reply:
If it were something done entirely inside `TouchSpeed`/`TouchAim`, I could reconsider, but this is fundamentally changing the mathematical basis of how strains are calculated from one strain to the next rather than changing the strain of a particular hitobject.

For example if you wanted to enforce something like "2 cursors" you could divide the screen into two halves and give a hitobject a strain of 0 if it occured in the opposite half as the last hitobject (or last-last object, or last-last-last object). That would be an implementation of this that I would be fine with, which wouldn't require changing the underlying `apply decay -> apply strain value -> get maximum` process.